modules/history/src/views/full/index.jsx

```diff
@@ -81,12 +85,13 @@ export default class FullView extends React.Component {
     this.getConversations(this.state.from, moment_day)
   }
 
-  fetchMoreMessages = async () => {
-    const { data } = await this.props.bp.axios.get(
-      `/mod/history/more-messages/${this.state.currentConversation}?offset=${
-        this.state.currentConversationMessageGroupsOffset
-      }&clientCount=${this.state.currentConversationMessageGroupsCount}`
-    )
+  fetchMoreMessages = async filters => {
+    let ressourceUrl = `${BASE_API_PATH}/more-messages/${this.state.currentConversation}?offset=${
+      this.state.currentConversationMessageGroupsOffset
+    }&clientCount=${this.state.currentConversationMessageGroupsCount}`
+    ressourceUrl += filters.flag ? '&flag=true' : ''
+
```

Comment:
That query looks complicated. Try to shorten it a bit. It's okay to take more space if it means that the code is more readable: 
```js
const convoId = this.state.currentConversation
const offset = this.state.currentConversationMessageGroupsOffset
const count = this.state.currentConversationMessageGroupsCount
const isFlagged = filters.flagged

const resourceUrl = "/mod/history/moremessages/${convoId}"

axios.get(resourceUrl, {params: { offset, count, isFlagged })